bundles/org.openhab.ui/web/src/components/widgets/chart/chart-mixin.js

```diff
@@ -0,0 +1,189 @@
+    },
+    singleAxis () {
+      if (!this.context.component.slots || !this.context.component.slots.singleAxis) return undefined
+      return this.context.component.slots.xAxis.map((a) => axisComponents[a.component].get(a, this.startTime, this.endTime, this.context.component))
+    },
+    tooltip () {
+      if (!this.context.component.slots || !this.context.component.slots.tooltip) return undefined
+      return this.context.component.slots.tooltip.map((c) => OhChartTooltip.get(c, this.startTime, this.endTime, this.$device))
+    },
+    visualMap () {
+      if (!this.context.component.slots || !this.context.component.slots.visualMap) return undefined
+      return this.context.component.slots.visualMap.map((c) => OhChartVisualMap.get(c, this.startTime, this.endTime, this.$device))
+    },
+    dataZoom () {
+      if (!this.context.component.slots || !this.context.component.slots.dataZoom) return undefined
+      return this.context.component.slots.dataZoom.map((c) => OhChartDataZoom.get(c, this.startTime, this.endTime, this.$device))
+    },
+    legend () {
+      if (!this.context.component.slots || !this.context.component.slots.legend) return undefined
+      return this.context.component.slots.legend.map((c) => OhChartLegend.get(c, this.startTime, this.endTime, this.$device))
+    },
+    title () {
+      if (!this.context.component.slots || !this.context.component.slots.title) return undefined
+      return this.context.component.slots.title.map((c) => OhChartTitle.get(c, this.startTime, this.endTime, this.$device))
+    },
+    toolbox () {
+      if (!this.context.component.slots || !this.context.component.slots.toolbox) return undefined
+      return this.context.component.slots.toolbox.map((c) => OhChartToolbox.get(c, this.startTime, this.endTime, this.$device))
+    }
+  },
+  asyncComputed: {
+    series () {
+      if (!this.context.component.slots || !this.context.component.slots.series) return Promise.resolve([])
+      return Promise.all(this.context.component.slots.series.map((s) => this.getSeriesPromises(s)))
+    }
+  },
+  methods: {
+    getSeriesPromises (component) {
+      const neededItems = seriesComponents[component.component].neededItems(component)
+      const promises = neededItems.filter(i => !!i).map((neededItem) => {
+        let url = `/rest/persistence/items/${neededItem}`
+        let seriesStartTime = this.startTime
+        let seriesEndTime = this.endTime
+        if (component.config.offsetAmount && component.config.offsetUnit) {
+          seriesStartTime = seriesStartTime.subtract(component.config.offsetAmount, component.config.offsetUnit)
+          seriesEndTime = seriesEndTime.subtract(component.config.offsetAmount, component.config.offsetUnit)
+        }
+        let query = {
+          serviceId: component.config.service || undefined,
```

Comment:
I see the service can be set here from the textual config , but is there anything in the GUI to let a user set this?  We could populate a list from http://oh/rest/persistence.